operator/src/test/java/org/keycloak/operator/testsuite/integration/UpgradeTest.java

```diff
@@ -102,6 +102,9 @@ public void testCacheMaxCount(UpdateStrategy updateStrategy) throws InterruptedE
     private static Keycloak createInitialDeployment(UpdateStrategy updateStrategy) {
         var kc = getTestKeycloakDeployment(true);
         kc.getSpec().setInstances(3);
+        // unable to use custom image, it fails with:
+        // The following build time options have values that differ from what is persisted - the new values will NOT be used until another build is run: kc.features
+        kc.getSpec().setImage(null); //unable to use custom image
```

Comment:
@ahus1 I understand the problem and why it fails. The GHA is building an optimized/custom image:

https://github.com/keycloak/keycloak/blob/006aea300d7892fd60cf756535f25d2c325b3d1c/operator/scripts/Dockerfile-custom-image#L5

And the container fails to start because I'm enabling the `rolling-updates` feature.
```
The following build time options have values that differ from what is persisted - the new values will NOT be used until another build is run: kc.features
```

Reply:
Yes, that explains it. Note that you can have a custom image and mark it as a non-optimized image by setting `startOptimized: false` in the spec.

Reply:
@ahus1 what is your preference for this test?
 `kc.getSpec().setImage(null)` or `startOptimized: false`